File: data/snippets/articles/guides/audio/basic/audio_basics.cpp
#include "splashkit.h"

int main() 
{
    sound_effect  snd_effect;

    open_window("Sound Demo", 320, 240);

    snd_effect = load_sound_effect("chipmunk", "chipmunk.ogg");

    load_sound_effect("bells", "bells.ogg");
    load_sound_effect("camera", "camera.ogg");
    load_sound_effect("boing", "comedy_boing.ogg");
    load_sound_effect("dinosaur", "dinosaur.ogg");
    load_sound_effect("bark", "dog_bark.ogg");

    do 
    {
        process_events();
        if ( key_down(RIGHT_CTRL_KEY) || key_down(LEFT_CTRL_KEY) )
        {
            if ( key_typed(NUM_1_KEY))
                snd_effect = sound_effect_named("chipmunk");
            if ( key_typed(NUM_2_KEY))
                snd_effect = sound_effect_named("bells");
            if ( key_typed(NUM_3_KEY))
                snd_effect = sound_effect_named("camera");
            if ( key_typed(NUM_4_KEY))
                snd_effect = sound_effect_named("boing");
            if ( key_typed(NUM_5_KEY))
                snd_effect = sound_effect_named("dinosaur");
            if ( key_typed(NUM_6_KEY))
                snd_effect = sound_effect_named("bark");
        }
        else
        {
            if ( key_typed(NUM_1_KEY))
                play_sound_effect(snd_effect);
            if ( key_typed(NUM_2_KEY))
                play_sound_effect(snd_effect, 0.5f);
            if ( key_typed(NUM_3_KEY))
                play_sound_effect(snd_effect, 3, 0.25);
            if ( key_typed(NUM_4_KEY))
                play_sound_effect(snd_effect, -1, 0.1);
            if ( key_typed(NUM_5_KEY))
            {
                if ( sound_effect_playing(snd_effect))
                    stop_sound_effect(snd_effect);
            }
        }

        clear_screen(COLOR_WHITE);

        draw_text("Control Sound (Escape or q to quit)", COLOR_RED, "arial", 18, 15, 15);
        draw_text("1: Play Sound At Full Volume", COLOR_BLUE, "arial", 14, 20, 50);
        draw_text("2: Play Sound At 50% Volume", COLOR_BLUE, "arial", 14, 20, 80);
        draw_text("3: Play Sound At 25% Volume 3 Times", COLOR_BLUE, "arial", 14, 20, 110);
        draw_text("4: Play Sound Continuously at 10%", COLOR_BLUE, "arial", 14, 20, 140);
        draw_text("5: Stop Playing Current Sound", COLOR_BLUE, "arial", 14, 20, 170);
        draw_text("CTRL + (1-6) switch sound effects", COLOR_BLUE, "arial", 14, 20, 200);

        refresh_screen(60);
    } while ( ! (quit_requested() || key_typed(ESCAPE_KEY) || key_typed(Q_KEY)));

    return 0;
}

File: data/snippets/articles/guides/audio/basic/audio_basics.cs
using SplashKitSDK;

class Program
{
    public static void Main()
    {
        SoundEffect sndEffect;

        SplashKit.OpenWindow("Sound Demo", 320, 240);

        sndEffect = new SoundEffect("chipmunk", "chipmunk.ogg");
        // or... 
        // sndEffect = SplashKit.LoadSoundEffect("chipmunk", "chipmunk.ogg");

        SplashKit.LoadSoundEffect("bells", "bells.ogg");
        SplashKit.LoadSoundEffect("camera", "camera.ogg");
        SplashKit.LoadSoundEffect("boing", "comedy_boing.ogg");
        SplashKit.LoadSoundEffect("dinosaur", "dinosaur.ogg");
        SplashKit.LoadSoundEffect("bark", "dog_bark.ogg");

        do
        {
            SplashKit.ProcessEvents();
            if (SplashKit.KeyDown(KeyCode.RightCtrlKey) || SplashKit.KeyDown(KeyCode.LeftCtrlKey))
            {
                if (SplashKit.KeyTyped(KeyCode.Num1Key))
                    sndEffect = SplashKit.SoundEffectNamed("chipmunk");
                if (SplashKit.KeyTyped(KeyCode.Num2Key))
                    sndEffect = SplashKit.SoundEffectNamed("bells");
                if (SplashKit.KeyTyped(KeyCode.Num3Key))
                    sndEffect = SplashKit.SoundEffectNamed("camera");
                if (SplashKit.KeyTyped(KeyCode.Num4Key))
                    sndEffect = SplashKit.SoundEffectNamed("boing");
                if (SplashKit.KeyTyped(KeyCode.Num5Key))
                    sndEffect = SplashKit.SoundEffectNamed("dinosaur");
                if (SplashKit.KeyTyped(KeyCode.Num6Key))
                    sndEffect = SplashKit.SoundEffectNamed("bark");
            }
            else
            {
                if (SplashKit.KeyTyped(KeyCode.Num1Key))
                    sndEffect.Play();
                if (SplashKit.KeyTyped(KeyCode.Num2Key))
                    sndEffect.Play(0.5f);
                if (SplashKit.KeyTyped(KeyCode.Num3Key))
                    sndEffect.Play(3, 0.25f);
                if (SplashKit.KeyTyped(KeyCode.Num4Key))
                    sndEffect.Play(-1, 0.1f);
                if (SplashKit.KeyTyped(KeyCode.Num5Key))
                {
                    if (sndEffect.IsPlaying)
                        sndEffect.Stop();
                }
            }

            SplashKit.ClearScreen(Color.White);

            SplashKit.DrawText("Control Sound (Escape or q to quit)", Color.Red, "arial", 18, 15, 15);
            SplashKit.DrawText("1: Play Sound At Full Volume", Color.Blue, "arial", 14, 20, 50);
            SplashKit.DrawText("2: Play Sound At 50% Volume", Color.Blue, "arial", 14, 20, 80);
            SplashKit.DrawText("3: Play Sound At 25% Volume 3 Times", Color.Blue, "arial", 14, 20, 110);
            SplashKit.DrawText("4: Play Sound Continuously at 10%", Color.Blue, "arial", 14, 20, 140);
            SplashKit.DrawText("5: Stop Playing Current Sound", Color.Blue, "arial", 14, 20, 170);
            SplashKit.DrawText("CTRL + (1-6) switch sound effects", Color.Blue, "arial", 14, 20, 200);

            SplashKit.RefreshScreen(60);
        } while (!(SplashKit.QuitRequested() || SplashKit.KeyTyped(KeyCode.EscapeKey) || SplashKit.KeyTyped(KeyCode.QKey)));
    }
}